cirq-core/cirq/ops/pauli_string_phasor.py

```diff
@@ -96,41 +132,54 @@ def exponent_relative(self) -> Union[int, float, sympy.Expr]:
         return self.gate.exponent_relative
 
     def _value_equality_values_(self):
-        return (self.pauli_string, self.exponent_neg, self.exponent_pos)
+        return (self.pauli_string, self.qubits, self.exponent_neg, self.exponent_pos)
 
     def equal_up_to_global_phase(self, other):
         """Checks equality of two PauliStringPhasors, up to global phase."""
         if isinstance(other, PauliStringPhasor):
-            rel1 = self.exponent_relative
-            rel2 = other.exponent_relative
-            return rel1 == rel2 and self.pauli_string == other.pauli_string
+            return (
+                self.exponent_relative == other.exponent_relative
+                and self.pauli_string == other.pauli_string
+                and self.qubits == other.qubits
+            )
         return False
 
     def map_qubits(self, qubit_map: Dict[raw_types.Qid, raw_types.Qid]):
         """Maps the qubits inside the PauliString."""
         return PauliStringPhasor(
-            self.pauli_string.map_qubits(qubit_map),
+            pauli_string=self.pauli_string.map_qubits(qubit_map),
+            qubits=[qubit_map[q] for q in self.qubits],
```

Comment:
This assumes that all `self.qubits` will be present as keys in `qubit_map`. Do we want to enforce this condition? If yes, let's update the docstring and raise a ValueError if qubit_map doesn't contain all the keys.

Reply:
done